File: PalindromeInteger.py
# Given an integer x, return true if x is a
# palindrome
# , and false otherwise.

import unittest

def isPalindrome(x):
    if x < 0:
        return False

    inputNum = x
    newNum = 0
    while x > 0:
        newNum = newNum * 10 + x % 10
        x = x // 10
    return newNum == inputNum

class IsPalindromeTestCase(unittest.TestCase):
    def test_palindrome_number(self):
        x = 121
        self.assertTrue(isPalindrome(x))

    def test_non_palindrome_number(self):
        x = 123
        self.assertFalse(isPalindrome(x))

    def test_negative_number(self):
        x = -121
        self.assertFalse(isPalindrome(x))

    def test_single_digit_number(self):
        x = 5
        self.assertTrue(isPalindrome(x))

    def test_zero(self):
        x = 0
        self.assertTrue(isPalindrome(x))

if __name__ == '__main__':
    unittest.main()

File: CTests/palindrome_integer.c
#include <stdio.h>
#include <stdbool.h>

// Function declaration
bool isPalindrome(int x);

// Function implementation
bool isPalindrome(int x) {
    if (x < 0) {
        return false;
    }

    int inputNum = x;
    int newNum = 0;
    while (x > 0) {
        newNum = newNum * 10 + x % 10;
        x = x / 10;
    }

    return newNum == inputNum;
}

// Test cases
void test_palindrome_number() {
    int x = 121;
    bool result = isPalindrome(x);

    printf("Test palindrome_number: ");
    if (result) {
        printf("PASSED\n");
    } else {
        printf("FAILED\n");
    }
}

void test_non_palindrome_number() {
    int x = 123;
    bool result = isPalindrome(x);

    printf("Test non_palindrome_number: ");
    if (!result) {
        printf("PASSED\n");
    } else {
        printf("FAILED\n");
    }
}

void test_negative_number() {
    int x = -121;
    bool result = isPalindrome(x);

    printf("Test negative_number: ");
    if (!result) {
        printf("PASSED\n");
    } else {
        printf("FAILED\n");
    }
}

void test_single_digit_number() {
    int x = 5;
    bool result = isPalindrome(x);

    printf("Test single_digit_number: ");
    if (result) {
        printf("PASSED\n");
    } else {
        printf("FAILED\n");
    }
}

void test_zero() {
    int x = 0;
    bool result = isPalindrome(x);

    printf("Test zero: ");
    if (result) {
        printf("PASSED\n");
    } else {
        printf("FAILED\n");
    }
}

int main() {
    test_palindrome_number();
    test_non_palindrome_number();
    test_negative_number();
    test_single_digit_number();
    test_zero();

    return 0;
}
